packages/controller/src/resources/app/index.ts

```diff
@@ -165,7 +167,7 @@ export function OpstraceApplicationResources(
   if (secretToCopy) {
     // use the value of the secret we want to copy
     secretValue = Buffer.from(
-      secretToCopy.data!.HASURA_ADMIN_SECRET,
+      secretToCopy.data?.HASURA_ADMIN_SECRET ?? "",
```

Comment:
This code section looks dangerous now, from a security point of view. Using a low-entropy secret like an empty string should be of no particular use/value -- not for testing, not in any dev environment, not in CI (and _obviously_ not in "prod").

I think security-wise I'd therefore love to be sure that `HASURA_ADMIN_SECRET` can never be an empty string.

That is, I'd love to be sure that we exit the process around the entry point when validating the config when `HASURA_ADMIN_SECRET` is not set to a string of the desired length.

After config validation, we can then rely on this invariant: `HASURA_ADMIN_SECRET` _is_ a secret of desired length (often implying desired entropy).

Deferring to @MatApple for his opinion, of course.

Reply:
Hasura rejects an empty value for admin secret (I tested to be also be sure), so we could validate earlier in the process if we want to throw an error before the downstream components do, which there is no harm in doing, but I'd say this is not an issue since the validation does indeed happen in the most important places. Our app will also throw an error if the value is empty and so will the controller.

Reply:
Good eye though @jgehrcke :-)